jetty-http/src/main/java/org/eclipse/jetty/http/MetaData.java

```diff
@@ -18,35 +18,44 @@
 
 package org.eclipse.jetty.http;
 
-import java.util.Collections;
 import java.util.Iterator;
 import java.util.function.Supplier;
 
 public class MetaData implements Iterable<HttpField>
 {
-    private HttpVersion _httpVersion;
-    private final HttpFields _fields;
-    private long _contentLength;
-    private Supplier<HttpFields> _trailers;
-
-    public MetaData(HttpVersion version, HttpFields fields)
+    private static final Supplier<HttpFieldList> SELF_SUPPLIED_TRAILORS = () -> null;
+    private static final HttpFieldList SUPPLIED_TRAILERS = new HttpFields().asImmutable();
+    private final HttpVersion _httpVersion;
+    private final HttpFieldList _fields;
+    private final long _contentLengthKnown;
+    private final long _contentLengthField;
+    private final Supplier<HttpFieldList> _trailerSupplier;
+    private HttpFieldList _trailers;
```

Comment:
Why isn't that field immutable?

Reply:
Because it is not part of the immutable meta-data.  rather it is a field of the self referencing trailerSupplier that is used if a specific supplier is not provided.
I will review this once I have trailers working and will then document a lot better if I keep it... I may move to an inner class if I can convince myself it wont be less efficient.